code/modules/mob/living/bot/ed209bot.dm

```diff
@@ -19,6 +29,201 @@
+			for(var/a in parsed_name)
+				if(a == "the" || length(a) < 2) //get rid of shit words.
+					continue
+				if(findtext(text,"[a]"))
+					found = 1
+					break
+		if(found)
+			return M
+	return null
+
+/mob/living/bot/secbot/ed209/proc/arrest_command(var/mob/speaker,var/text)
+	target = null //want me to attack something? Well I better forget my old target.
+	walk_to(src,0)
+
+	target = get_target_by_name(text)
+	if(!(target in view(7, src)))
+		return 0
+
+	if(findtext(text,"detain"))
+		arrest_type = 1
+	else
+		arrest_type = 0
+
+	if(isnull(target))
+		custom_emote(2, "[emote_hear], \"Error, unit is unable to find target in view range!\"")
+		return 0
+	else
+		if(ishuman(target))
+			idcheck = TRUE
+			var/mob/living/carbon/human/H = target
+			var/perpname = H.name
+			var/obj/item/weapon/card/id/id = H.GetIdCard()
+			if(id)
+				perpname = id.registered_name
+
+			var/datum/data/record/R = find_security_record("name", perpname)
+			if(R)
+				R.fields["criminal"] = "*Arrest*"
+			else
+				custom_emote(2, "[emote_hear], \"Warning, [target] does not have Security records! Enabling security records check mode!\"")
+				check_records = TRUE
+			mode = SECBOT_HUNT
+			custom_emote(2, "[emote_hear], \"[arrest_type ? ("Detaining") : ("Arresting")] [target]\"")
+
+/mob/living/bot/secbot/ed209/proc/stay_command(var/mob/speaker,var/text)
+	walk_to(src, src, 0, move_to_delay)
+	mode = SECBOT_IDLE
+	auto_patrol = 0
+	target = null
+	check_records = FALSE
+	custom_emote(2, "[emote_hear], \"Roger that, going into idle mode. Auto patrol disabled.\"")
+	return 1
+
+/mob/living/bot/secbot/ed209/proc/stop_command(var/mob/speaker,var/text)
+	if(!on)
+		return
+	walk_to(src, src, 0, move_to_delay)
+	check_records = FALSE
+	custom_emote(2, "[emote_hear], \"Roger that, unit going offline.\"")
+	turn_off()
+	return 1
+
+/mob/living/bot/secbot/ed209/proc/patrol_command(var/mob/speaker,var/text)
+	walk_to(src, src, 0, move_to_delay)
+	mode = SECBOT_IDLE
+	auto_patrol = 1
+	target = null
+	custom_emote(2, "[emote_hear], \"Roger that, starting patrol now.\"")
+	return 1
+
+/mob/living/bot/secbot/ed209/proc/follow_command(var/mob/speaker,var/text)
+	//we can assume 'stop following' is handled by stop_command
+	if(findtext(text,"me"))
+		mode = SECBOT_FOLLOW
+		target = speaker
+		custom_emote(2, "[emote_hear], \"Roger that, following you\"")
+		return 1
+
+	target = get_target_by_name(text)
+	if(!(target in view(7, src)))
+		return 0
+
+	if(!target)
+		return 0
+
+	mode = SECBOT_FOLLOW
+	custom_emote(2, "[emote_hear], \"Roger that, following [target]\"")
+
+	return 1
+
+/mob/living/bot/secbot/ed209/verb/change_name()
+	set name = "Change nickname"
+	set category = "IC"
+	set src in view(1)
+
+	var/mob/M = usr
+	if(!M.mind)	return 0
+
+	var/short_input = sanitizeSafe(input("What do you want [src] respond to?", , ""), MAX_NAME_LEN)
```

Comment:
Minor nitpick, but I don't see any restrictions for changing the nickname of the ed209 unit. Would a criminal be able to change the nickname of the ED-209 unit to something else?

Reply:
That is a good point